tests/test_sequence.py

```diff
@@ -137,6 +137,221 @@ def test_magnetic_field():
+                    2 * np.pi * 10,
+                    clock_period=3,
+                    phase_jump_time=500,
+                    max_duration=2**26,
+                    max_targets=5,
+                ),
+            ),
+        ),
+    )
+
+    test_device3 = Device(
+        name="test_device1",
+        dimensions=2,
+        rydberg_level=50,
+        max_atom_num=100,
+        max_radial_distance=60,
+        min_atom_distance=5,
+        _channels=(
+            (
+                "rmn_local1",
+                Raman.Local(
+                    2 * np.pi * 20,
+                    2 * np.pi * 10,
+                    clock_period=4,
+                    phase_jump_time=750,
+                    max_duration=2**26,
+                    max_targets=6,
+                ),
+            ),
+            (
+                "rmn_local2",
+                Raman.Local(
+                    2 * np.pi * 20,
+                    2 * np.pi * 10,
+                    clock_period=1,
+                    phase_jump_time=500,
+                    max_duration=2**26,
+                    max_targets=1,
+                ),
+            ),
+            (
+                "rmn_local3",
+                Raman.Local(
+                    2 * np.pi * 20,
+                    2 * np.pi * 10,
+                    clock_period=2,
+                    phase_jump_time=500,
+                    max_duration=2**26,
+                    max_targets=2,
+                ),
+            ),
+            (
+                "rydberg_global",
+                Rydberg.Global(
+                    max_abs_detuning=2 * np.pi * 4,
+                    max_amp=2 * np.pi * 3,
+                    phase_jump_time=500,
+                    max_duration=2**26,
+                ),
+            ),
+        ),
+    )
+
+    reg = Register.square(3, 5, prefix="q")
+
+    # Device checkout
+    seq = Sequence(reg, Chadoq2)
+    with pytest.warns(
+        UserWarning,
+        match="Switching a sequence to the same device"
+        + " returns the sequence unchanged.",
+    ):
+        seq.switch_device(Chadoq2)
+
+    seq = Sequence(reg, Chadoq2)
+    with pytest.raises(
+        NotImplementedError,
+        match="Switching the device of a sequence to one"
+        + " with reusable channels is not supported.",
+    ):
+        seq.switch_device(MockDevice)
+
+    # Different Channels basis
+    seq = Sequence(reg, test_device1)
+    seq.declare_channel("digital", "raman_global", "q1")
+
+    with pytest.raises(TypeError, match="No match for channel digital."):
+        seq.switch_device(IroiseMVP)
+
+    # Different addressing channels
+    seq = Sequence(reg, test_device1)
+    seq.declare_channel("ising", "raman_global")
+
+    with pytest.raises(TypeError, match="No match for channel ising."):
+        seq.switch_device(test_device2)
+
+    seq = Sequence(reg, MockDevice)
+    seq.declare_channel("ising", "rydberg_global")
+    with pytest.warns(UserWarning):
```

Comment:
What warning is this?

Reply:
It's the one about the change on the Rydberg level

Reply:
You should specify it here

Reply:
I was stick because when there are two warnings, I get a list of them, so that the macthing doesn't work as usual.

Reply:
What other warning are you getting here?

Reply:
![image](https://user-images.githubusercontent.com/76184755/203015375-e3f73c31-bbdc-4fb1-a3ec-73d735959a29.png)

This is the error I met

Reply:
Well, it seems that the warning messages don't match. This might be because of regex special characters. Add `import re`at the top of the file and then put your match string inside `re.escape()`. Let's see if that works.